### ISSUE TITLE
Don't delete target folders on cleaning

### ISSUE THREAD

**Issue by ceeliro** [Alfresco/alfresco-sdk#261]
Would be possible to delete only the "target" folders content when running cleaning task? For those that are using eclipse it is a hassle each time that we have to mark these folders as derived after cleaning.

But it is only a request not and issue since it is do able through the poms.

All best, thank for the good job.

**Comment by gravitonian** [comment]
Hi,

You can amend the maven-clean-plugin in your top/parent pom to do what you want, add this profile to profiles:

       <profile>
            <id>purge</id>
            <build>
                <plugins>
                    <plugin>
                        <artifactId>maven-clean-plugin</artifactId>
                        <configuration>
                            <excludeDefaultDirectories>true</excludeDefaultDirectories>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>
This makes the clean plugin ignore the default cleaning folders (i.e. /target)

**Comment by mindthegab** [comment]
Also:

are you using by any chance Eclipse Luna (which has m2eclipse in it)? Asking since I have never had to re-mark folders as derived, m2eclipse take care of syncing the maven and eclipse build lifecycles. 

One addition wrt @gravitonian's comment: deleting target folders is part of the standard Maven lifecycle (nothing SDK specific), so I don't think what you really want is to to excludeDefaultDirectories (I assume you want to delete only the contents of /target, not the /target folder itself). You might technically do that manipulating [filesets|http://maven.apache.org/plugins/maven-clean-plugin/clean-mojo.html#filesets] on top of excludeDefaultDirectories, but again I suggest you double check you m2eclipse functionality as you might have an issue there.

Closing this one as not really SDK related.

**Comment by ceeliro** [comment]
Thanks for the answer... I followed your advice! It is true that it is not really an SDK issue but maven behaviour...

Thanks

**Comment by stokito** [comment]
If someone like me looking for the answer:
```xml
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-clean-plugin</artifactId>
                <version>3.0.0</version>
                <executions>
                    <execution>
                        <id>Deleting all files under target, but not target itself</id>
                        <phase>clean</phase>
                        <goals>
                            <goal>clean</goal>
                        </goals>
                    </execution>
                </executions>
                <configuration>
                    <excludeDefaultDirectories>true</excludeDefaultDirectories>
                    <filesets>
                        <fileset>
                            <directory>target/</directory>
                            <followSymlinks>false</followSymlinks>
                            <includes>
                                <include>**/*</include>
                            </includes>
                        </fileset>
                    </filesets>
                </configuration>
            </plugin>
```

**Comment by Diarsid** [comment]
@stokito 
I've been looking for an ability to delete all /targets content but not target directory itself for a while and I've tried this approach. I've found out that it doesn't work as one could expect. This configuration deletes all, except /target/classes, /target/test-classes and all their content respectively. They remain completely unaffected by cleaning. 

It seems that it is a maven flaw. I've tried different combination of <inlude>...</include> element and couldn't achieve desired behavior.

Maven version I'm currently using is 3.3.3. If I remember correctly a previous Maven version I used was 3.2.2 and it had the same fault.
